src/App.tsx

```diff
@@ -1,194 +1,149 @@
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterValue, setFilterValue] = useState('all');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todo => setTodos(todo))
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  }, []);
+
+  const visibleTodos = filterBy([...todos], filterValue);
+  const activeItems = todos.filter(todo => !todo.completed);
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
-
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="TodosCounter">
-            3 items left
-          </span>
-
-          {/* Active filter should have a 'selected' class */}
-          <nav className="filter" data-cy="Filter">
-            <a
-              href="#/"
-              className="filter__link selected"
-              data-cy="FilterLinkAll"
-            >
-              All
-            </a>
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
 
-            <a
-              href="#/active"
-              className="filter__link"
-              data-cy="FilterLinkActive"
-            >
-              Active
-            </a>
+      <div className="todoapp__content">
+        <Header
+          setTodos={setTodos}
+        />
+
+        <TodoList
+          todos={visibleTodos}
+          setTodos={setTodos}
+        />
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {activeItems.length ? (
+                `${activeItems.length} items left`
+              ) : (
+                '0 items left'
+              )}
+            </span>
 
-            <a
-              href="#/completed"
-              className="filter__link"
-              data-cy="FilterLinkCompleted"
+            {/* Active filter should have a 'selected' class */}
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={cn('filter__link', {
+                  selected: filterValue === 'all',
+                })}
+                data-cy="FilterLinkAll"
+                onClick={() => {
+                  setFilterValue('all');
+                }}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={cn('filter__link', {
+                  selected: filterValue === 'active',
+                })}
```

Comment:
you can also create the footer component

Reply:
good job, but i suggest to create the footer component, it will simplify the app.tsx